protobuf-kafkaconnect-converter/src/test/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/ProtobufSchemaConverterTest.java

```diff
@@ -157,4 +170,64 @@ public void toConnectData_convertsProtobufSerializedDataToConnectData(
         assertEquals(expectedSchemaAndValue, schemaAndValue);
     }
+
+    @Test
+    public void endToEndTest_forAllTypesSchema() throws IOException {
+
+        Object connectData = ToProtobufTestDataGenerator.getAllTypesData("AllTypes");
+        Schema connectSchema = ToProtobufTestDataGenerator.getAllTypesSchema("AllTypes");
+        DynamicMessage protobufData = ToProtobufTestDataGenerator.getProtobufAllTypesMessage("AllTypes");
+
+        // from connect
+        ArgumentCaptor<Message> argumentCaptor = ArgumentCaptor.forClass(Message.class);
+        doReturn(new byte[] {}).when(serializer).serialize(eq(TOPIC_NAME), any());
+        protobufSchemaConverter.fromConnectData(TOPIC_NAME, connectSchema, connectData);
+        verify(serializer, times(1)).serialize(eq(TOPIC_NAME), argumentCaptor.capture());
+
+        // assert that ConnectToProtobuf conversion is correct
+        assertEquals(protobufData.toString(), argumentCaptor.getValue().toString());
+
+        ProtobufSerializer protobufSerializer =
+                new ProtobufSerializer(new GlueSchemaRegistryConfiguration(new HashMap<String, String>() {{
+                    put(AWSSchemaRegistryConstants.AWS_REGION, "us-west-2");
+                }}));
+
+        // serializedData is the raw data from the serialization
+        final byte[] serializedData = protobufSerializer.serialize(protobufData);
+
+        // fileDescriptor is the schema definition reconstructed from connect data and is registered with Glue Schema
+        // Registry service
+        Descriptors.FileDescriptor fileDescriptor = new ConnectSchemaToProtobufSchemaConverter().convert(connectSchema);
+        ProtobufWireFormatDecoder protobufWireFormatDecoder = new ProtobufWireFormatDecoder(new MessageIndexFinder());
+        DynamicMessage deserializedMessage =
+                (DynamicMessage) protobufWireFormatDecoder.decode(serializedData, fileDescriptor,
+                        ProtobufMessageType.DYNAMIC_MESSAGE);
+        // Mockito is used to mock the actual deserialization result
+        doReturn(deserializedMessage).when(deserializer).deserialize(TOPIC_NAME, serializedData);
+
+        // to connect
+        SchemaAndValue schemaAndValue = protobufSchemaConverter.toConnectData(TOPIC_NAME, serializedData);
+
+        // assert that end to end conversion result is the same as the input
+        Schema actualSchema = schemaAndValue.schema();
+        Object actualData = schemaAndValue.value();
+        assertEquals(connectSchema.name(), actualSchema.name());
+        assertEquals(connectSchema.fields().size(), actualSchema.fields().size());
+        for (Field field: connectSchema.fields()) {
+            String fieldName = field.name();
+            Field actualField = actualSchema.field(fieldName);
+            // assert each field schema is the same
+            assertEquals(field, actualField, fieldName);
+
+            Object expectedFieldObject = ((Struct) connectData).get(field);
```

Comment:
Do we specifically assert for struct and map types?

Reply:
The assertion here applies to all the fields including both struct and map types. And both struct and map types have passed the assertion for both the schema and value. I don't think other assertions are needed.